src/ansys/fluent/core/services/field_data.py

```diff
@@ -838,6 +838,121 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
     return fields_data
 
 
+class _FieldDataBase:
+    def __init__(self, i_d, data):
+        self._data = data
+        self._id = i_d
+
+    @property
+    def data(self):
+        return self._data
+
+    @property
+    def surface_id(self):
+        return self._id
+
+    @property
+    def size(self):
+        return len(self._data)
+
+    def __getitem__(self, item):
+        return self._data[item]
+
+
+class ScalarFieldData(_FieldDataBase):
+    class _ScalarData:
+        def __init__(self, data):
+            self.scalar_data = data
+
+    def __init__(self, i_d, data):
+        super().__init__(i_d, [ScalarFieldData._ScalarData(_data) for _data in data])
+
+
+class Vector:
```

Comment:
@prmukherj , In a follow-up task - when you have time - you could add docstrings.

Reply:
I'll add docstrings as part of this PR only.

Reply:
Done, Thank you.